### ISSUE TITLE
Unable to load GDExtension in Web export due to LinkError with env.strtoll

### ISSUE THREAD

**Issue by Nihlus** [godotengine/godot#96492]
### Tested versions

- Reproducible in 4.3.stable.official.77dcf97d8

### System information

Godot v4.3.stable - Linux Mint 22 (Wilma) - X11 - Vulkan (Forward+) - dedicated Radeon RX Vega (RADV VEGA10) - AMD Ryzen 9 7950X 16-Core Processor (32 Threads)

### Issue description

I have a GDExtension project that statically links with a relatively large speech recognition library. It works fine on desktop platforms but fails to load in Web exports, showing a LinkError from the WASM runtime.

```
Uncaught (in promise) LinkError: imported function 'env.strtoll' signature mismatch 
```

This issue also presents when using symbols from `std::chrono` in the extension, even though none of them are exported or made available to other modules. The problem is described in https://github.com/godotengine/godot/issues/94537#issuecomment-2239297532, and I am experiencing more or less the same thing.

### Steps to reproduce

1. Extract MRP
2. Clone godot-cpp into the extracted folder using the 4.3 branch
3. build with scons
4. export a web version of the project
5. run it
6. observe error in dev console

### Minimal reproduction project (MRP)

[minimal-repro.zip](https://github.com/user-attachments/files/16840835/minimal-repro.zip)

If you want to test the full project or the MRP does not work for whatever reason, the full source code is available at https://github.com/Nihlus/gdvosk. Binaries can be downloaded from the workflows as artifacts. I had some trouble running the MRP due to what appears to be another Godot bug regarding stack smashing on my system, so YMMV.

**Comment by Nihlus** [comment]
I believe I found the problem - there's a compilation flag mismatch between godot-cpp and godot proper, meaning that `-sWASM_BIGINT` is not present (among a number of other flags). This flag changes the representation of 64-bit integers in the compiled code, resulting in types like `long long` having different signatures between the builds.

The solution is to add this flag to the compilation, but there should probably be a review of the rest of the flags as well. There's also some discrepancies when using the CMake support in godot-cpp, but that's probably better as a separate issue.

**Comment by Faless** [comment]
Thank you for looking into this, we forgot to add WASM_BIGINT to the godot-cpp build when we added it to godot in #88594 .

Should be fixed once godotengine/godot-cpp#1566 is merged.
